### PR TITLE
✨ Switch to numerical distance accuracy

### PR DESCRIPTION
**Pull request by niklas-heer**
This is picking up on #78

### CODE REVIEW

**Comment by francescoalemanno** [comment]
The formula makes no sense, mathematically speaking.
image
use our formulas

File: scmeta/src/scmeta.cr [hunk 1]
```diff
@@ -104,12 +100,24 @@ hyperfine = File.open("#{hyperfine_file}") do |file|
   JSON.parse(file).on_steroids!
 end
 
+def pi_accuracy(input : String)
+  # https://www.wolframalpha.com/input?i=N%5BPi%2C+32%5D
+  const_pi = BigFloat.new(3.1415926535897932384626433832795)
```
Comment (giordano) [inline scmeta/src/scmeta.cr:105]: I'm not familiar with Crystal, but if it's like julia then the conversion of a 64-bit floating point literal number to an arbitrary precision number is only going to give you garbage after the ~16th significant digit, at which point you can just stick to plain 64-bit floating point numbers

File: scmeta/src/scmeta.cr [hunk 2]
```diff
@@ -104,12 +100,24 @@ hyperfine = File.open("#{hyperfine_file}") do |file|
...
+  pi = input.to_big_f
+
+  if pi > const_pi
+    accuracy = 1 - ( pi / const_pi)
```
Comment (francescoalemanno) [inline scmeta/src/scmeta.cr:109]: change it to

accuracy = 2 - ( pi / const_pi)

File: scmeta/src/scmeta.cr [hunk 3]
```diff
@@ -104,12 +100,24 @@ hyperfine = File.open("#{hyperfine_file}") do |file|
...
+  else
+    accuracy = ( pi / const_pi)
+  end
+  accuracy.abs.to_f
```
Comment (francescoalemanno) [inline scmeta/src/scmeta.cr:113]: instead of abs, set it to 0 if negative

File: scmeta/src/scmeta.cr (hunk 1 above)
Comment (giordano) [reply scmeta/src/scmeta.cr:105]: And judging by
image
I believe I'm right.

**Review by francescoalemanno — CHANGES_REQUESTED** [review]
with the requested changes it is acceptable

image

File: scmeta/src/scmeta.cr (hunk 1 above)
Comment (niklas-heer) [reply scmeta/src/scmeta.cr:105]: @giordano fair point. It doesn't even seem to help to [set the precision](https://crystal-lang.org/api/1.6.1/BigFloat.html#new%28num%3AFloat%2Cprecision%3AInt%29-class-method).
grafik

That's probably why the [constant they have uses 16 digits](https://crystal-lang.org/api/1.6.1/Math.html). Then I'll be using that.

Comment (giordano) [reply scmeta/src/scmeta.cr:105]: If anything, you should use the [string constructor](https://crystal-lang.org/api/1.6.1/BigFloat.html#new%28str%3AString%29-class-method), like you do at the line below for the output of the program.

File: scmeta/src/scmeta.cr (hunk 3 above)
Comment (niklas-heer) [reply scmeta/src/scmeta.cr:113]: I'll use what you suggested instead.

**Comment by niklas-heer** [comment]
Thank you, both @francescoalemanno and @giordano for your valuable input 🙏

**Comment by niklas-heer** [comment]
Interesting results 😅
combined_results

[combined_results.csv](https://github.com/niklas-heer/speed-comparison/files/9874592/combined_results.csv)

**Pull request merged**